### ISSUE TITLE
Vectorization of functions using diffrax with xmap

### ISSUE THREAD

**Issue by nikithiel** [patrick-kidger/diffrax#246]
Hello there,

I'm using diffrax in combination with vmap in my project. This works like a charm. As my input data which I'm using for mapping gets bigger and bigger I want to enable parallel computing. As I wan't my code and the dimension of the array which is used for mapping being independent of the available number of logical devices, I thought of using [xmap](https://jax.readthedocs.io/en/latest/_autosummary/jax.experimental.maps.xmap.html). Unfortunately, I don't get it to work having the following error:

`ValueError: body_fun must have the same input and output structure`

I created a MVE based on one of your simple examples which shows how I intend to use vmap/xmap.

```
from diffrax import diffeqsolve, Dopri5, ODETerm, SaveAt, PIDController
import jax.numpy as jnp
from jax import vmap, pmap
from jax.experimental.maps import xmap

def wrapperDiffEq(y0):
    `vector_field = lambda t, y, args: -y
    term = ODETerm(vector_field)
    solver = Dopri5()
    saveat = SaveAt(ts=[0., 1., 2., 3.])
    stepsize_controller = PIDController(rtol=1e-5, atol=1e-5)
    sol = diffeqsolve(term, solver, t0=0, t1=3, dt0=0.1, y0=y0, saveat=saveat,
                      stepsize_controller=stepsize_controller)
    return sol

y0 = jnp.array([1, 2, 3])

#sol = vmap(wrapperDiffEq, in_axes=(0))(y0)

sol = xmap(wrapperDiffEq, in_axes=(['init', ...]), out_axes=(['init']))(y0)

print(sol.ts[0])  # DeviceArray([0.   , 1.   , 2.   , 3.    ])
print(sol.ys)  # DeviceArray([1.   , 0.368, 0.135, 0.0498])
```

I think xmap introducing [named_axis](https://jax.readthedocs.io/en/latest/notebooks/xmap_tutorial.html) might be the problem. Has any of you ever used diffrax in combination with xmap? And does any of you have a solution to my problem?

Thanks in advance and all the best
Niklas

**Comment by patrick-kidger** [comment]
Thanks for the bug report! Looks like something in Equinox needed tweaking to work with xmap. This should be fixed in https://github.com/patrick-kidger/equinox/pull/328.

(After that fix, I also had to change `out_axes` to `['init', ...]` in order to run your script.)

**Comment by nikithiel** [comment]
Hey Patrick, thanks for your help and fixing this!

**Comment by patrick-kidger** [comment]
Closing, as this is fixed in Equinox v0.10.3.
